### DISCUSSION TITLE
Help needed for mikemucc / screenlogic-api

### DISCUSSION THREAD

**Discussion by ivanvach** [tagyoureit/nodejs-poolController#654] [App and O/S Config Questions]
After trying nodejs-poolController (all 3 components installed and tested) I realized it is not for me as it is targeting mostly DIY enthusiasts for building pool controller from scratch. 
I'm not there yet as I have 4 fully functional Pentair controllers to control my pool and water features.
I also have 2 Pentair Screenlogic adapters that open the Pentair controllers to the network.
The problem is that  Pentair Screenlogic app is closed and doesn't have an API to open it to my home automation.
Hence my strong interest in mikemucc / screenlogic-api

I managed to install and run it on Raspberry Pi. I also set the "Enviroment Variables".

No errors in the Raspberry Pi console. On the contrary a brief console report says that it found my Screenlogic adapter and displayed status of 8 pumps . So far so good.

However when I try to use any of the GET endpoints via browser I'm getting "no found" error in the browser.

http://192.168.1.149:82/api/all

Please help if you can.

**Comment by tagyoureit** [comment]
Hi @ivanvach ,
this project (nodejs-poolController) was born out of controlling EasyTouch, and grew to include Intellitouch, Intellicenter, and many other pieces of equipment (both Pentair and others).  All you need is an RS-485 adapter and you will immediately get full control of your Pentair systems.  The ability to completely build your own controller from scratch has been a growing part of this project but you need not start there or ever go there.  If you just want to connect/control your Pentair systems you only need an [RS-485 controller](https://github.com/tagyoureit/nodejs-poolController/wiki/RS-485-Adapter-Details) (a few suggestions on this page, but any will work).  

This project isn't related to mikemucc/screenlogic-api.  However, I have just integrated the underlying project from parnic/node-screenlogic and I have a beta ready for testing.  You can definitely try this, but my recommendation is the rs-485 adapter as it is fully compatible with your equipment (and then, extendable, if you desire).

**Comment by ivanvach** [comment]
Hi @tagyoureit ,

Thank you very much for the detailed response. 
I do have RS-485 adapter Gearmo brand as recommended.
However I do have also 2 (two) Pentair adapters that have RS-485 build-in 
https://www.poolweb.com/products/interface-pc-itc15-intellitouch

Why piling up equipment on top of other equipment. The reliability drops as number of equipment goes up.

Furthermore, I couldn't find simple interface and protocol whereby I can issue simple commands like this

Aux21=on
Pool=on
SPA=off
Lights=on
Feature5=on

My home automation system is HomeSeer and I have Big6 plug-in that provides various interfaces. I can do MQTT, TCP, HTTP, UDP, Chat (Telegram or Slack) or Serial (including RS-485).

As a matter of fact if I would know the native RS-485 commands than I can issue them directly, but it means to get into your turf that I have no desire doing. It's a lot of work and you're doing a great job. Respect.

Bottom line please show me a way to manage my two systems (each with two Pentair Intellitouch) with simple commands as above examples over any interface (MQTT, TCP ... you choose) with or without RS-485 hardware.

**Reply by tagyoureit** [reply]
> Why piling up equipment on top of other equipment. The reliability drops as number of equipment goes up.

nodejs-poolController will run on Windows, Linux, Mac but none of those know how to talk RS-485.  The RS-485 bus is designed to handle many pieces of equipment at the same time.  We have never had any issues because someone added another device on the bus.  And you can wire into the OCP (outdoor control panel) or directly into the Screenlogic module.  

I suggest you start with getting njsPC up and running.   You can start with the [dashPanel](https://github.com/rstrouse/nodejs-poolController-dashPanel) UI to make sure everything is connecting to your pool as expected.  Once that is up and running, you can setup the [HomeSeer](https://github.com/tagyoureit/nodejs-poolController/wiki/Homeseer-Setup-Instructions) integration.  There are also HTTP, MQTT, HomeAssistant, Vera, etc [integrations](https://github.com/tagyoureit/nodejs-poolController/wiki/Bindings-Integrations-in-2.0).

The aforementioned Screenlogic integration would replace the RS-485 equipment above and talk directly to Screenlogic over TCP, but it is not as full featured as directly talking on the RS-485 bus.

**Comment by ivanvach** [comment]
@tagyoureit 

Understood and agreed on the equipment. As a side note RS-485 could be tricky sometimes especially with multiple pumps and equipment. I needed to play with end of line termination to address reflections and needed to add RC (resistor-capacitor) pairs at Pentair distribution board to better match the impedance.

I looked at Homeseer integration and I don't like it at all. It is well discussed at Homeseer forums and I'm not the only one who doesn't like it.

1. Is there a way for me to control directly using your software and Rs-485 and any interface TCP, MQTT etc. 

2. How far is the beta of the Screenlogic integration. There is an extra benefit for me using it as I do have two systems at two locations. If I go Rs-485 route than I will need two of everything Raspberry Pi, RS 485 adapter etc. installed at poolside locations that are not very accommodating. If I go with your Screenlogic integration than I can have everything at better location and only one RPi running the show. Furthermore, all I need to control is very basic.

Pool, SPA, AUX , Features and Lights.

I don't care about chlorinators, chemistry pumps, autofill, valves etc.

Please advise.

Best regards,

Ivan

**Reply by tagyoureit** [reply]
1.  Yes, There is an MQTT interfaces, API's, sockets.  They are in the `integrations` link I posted above.  Most of the [API's](https://github.com/tagyoureit/nodejs-poolController-api) are listed here, but I haven't kept the docs up to date so it's good to look at the dashPanel console to see if anything is updated.

2.  It's pretty far.  Most everything has been tested on the system control side.  There is more work to do on the configuration side but that is generally less important.

**Reply by ivanvach** [reply]
@tagyoureit 

Thanks.

2. Yes I have it working. See the other thread dedicated to it. I posted screenshots there and feedback about minor issues that I found. I also asked some questions if you’d be so kind to answer.
